### PR TITLE
feat(STADTPULS-355): user tokens page

### PR DESCRIPTION
**Pull request by dnsos**
This PR adds the page where users manage their tokens.

---
~~Note that only minimal tests for the `useUserTokens` hook are included in this PR. I'm having some issues getting the SWR stuff to work in the test setup. @vogelino, perhaps we can have a chat about it, and/or tackle this separately in another PR.~~

---

PS: GitHub is listing some commits here that actually don't belong to this PR. The "Files changed" are correct, however.

### CODE REVIEW

File: src/components/UserInfoHeader/withData.tsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ export const UserInfoWithData: FC<UserInfoWithDataPropType> = ({
               tabs={tabs}
             />
           </div>
-          {isOwnerAndLoggedIn && (
+          {isOwnerAndLoggedIn && activeTab === "sensors" && (
```
Comment (vogelino) [inline src/components/UserInfoHeader/withData.tsx:153]: Good catch! I forgot this! 👍

**Review by vogelino — APPROVED** [review]
Very clean implementation! Well tested (except for the getter, which we should tackle together) and very readable! 

I left a few "meckern auf hohem Niveau" notes which you can happily ignore. :)

File: src/components/Alert/index.tsx [hunk 2]
```diff
@@ -18,29 +19,36 @@ const getStylesByType = (type: AlertPropType["type"]): string =>
     .filter(Boolean)
     .join(" ");
 
-export const Alert: FC<AlertPropType> = ({ title, type = "info", message }) => {
+export const Alert: FC<AlertPropType> = ({
+  title,
+  type = "info",
+  message,
+  isRemovable = true,
```
[the remark this answers is not in the document]
Comment (dnsos) [reply src/components/Alert/index.tsx:26]: Good idea. I added a test and a Storybook entry in 18d4be3.

File: pages/accounts/[username]/tokens.tsx [hunk 3]
```diff
@@ -47,12 +74,76 @@ const AccountTokensPage: FC<AccountTokensPagePropType> = ({ routeAccount }) => {
...
+      {isOwnerAndLoggedIn && (
+        <div className='container max-w-8xl mx-auto px-4 py-16'>
+          <TokenForm
+            onSubmit={async tokenDescription => {
+              const tokenResponse = await createToken(tokenDescription);
+              setNewToken(tokenResponse.data.token);
+              setNewTokenDescription(tokenDescription);
+            }}
+          />
+          {error && (
```
[the remark this answers is not in the document]
Comment (dnsos) [reply pages/accounts/[username]/tokens.tsx]: Of course, perfect use and I'm wondering why I haven't thought about that earlier...

I've added the Alert displaying the error now.

File: pages/accounts/[username]/tokens.tsx [hunk 4]
```diff
@@ -82,18 +82,29 @@ const AccountTokensPage: FC<AccountTokensPagePropType> = ({ routeAccount }) => {
       )}
       {isOwnerAndLoggedIn && (
         <div className='container max-w-8xl mx-auto px-4 py-16'>
+          {error && (
+            <div className='mb-8'>
+              <Alert
```
Comment (vogelino) [inline pages/accounts/[username]/tokens.tsx:87]: Great thx! 👍

**Comment by dnsos** [comment]
Hey @vogelino, thanks for the review and the support for writing the tests!

I have now added tests for each token utility individually:

- `getUserTokens`
- `createUserToken`
- `regenerateUserToken`
- `deleteUserToken`

Can you please take a quick look again?

File: src/lib/hooks/useUserTokens/index.ts [hunk 5]
```diff
@@ -5,13 +5,13 @@ import useSWR, { mutate } from "swr";
...
-import { deleteUserToken } from "./deleteUserToken";
-import { getUserTokens } from "./getUserTokens";
+} from "../../requests/createUserToken";
+import { deleteUserToken } from "../../requests/deleteUserToken";
+import { getUserTokens } from "../../requests/getUserTokens";
 import {
   RegenerateTokenResponseSignature,
   regenerateUserToken,
-} from "./regenerateUserToken";
+} from "../../requests/regenerateUserToken";
```
Comment (vogelino) [inline src/lib/hooks/useUserTokens/index.ts:14]: (optional): Use absolute paths like `@lib/requests/...`

**Review by vogelino — APPROVED** [review]
Great testing! Thx! This feels very robust now! :)

**Comment by dnsos** [comment]
Great, thanks!

I will ask the team to do some more manual testing before merging.

**Review by vogelino — APPROVED** [review]
Thx for the fixes! Well done!

**Pull request merged**